api/groups/nodeGroup.go

```diff
@@ -241,6 +249,37 @@ func (ng *nodeGroup) peerInfo(c *gin.Context) {
 	)
 }
 
+// peerInfo returns the information of a provided p2p peer ID
+func (ng *nodeGroup) epochStartDataForEpoch(c *gin.Context) {
+	epoch, err := getQueryParamEpoch(c)
+	if err != nil {
+		shared.RespondWithValidationError(c, errors.ErrValidation, errors.ErrBadUrlParams)
+		return
+	}
+
+	epochStartData, err := ng.getFacade().GetEpochStartDataForEpoch(epoch)
+	if err != nil {
+		c.JSON(
+			http.StatusInternalServerError,
+			shared.GenericAPIResponse{
+				Data:  nil,
+				Error: fmt.Sprintf("%s: %s", errors.ErrGetEpochStartData.Error(), err.Error()),
+				Code:  shared.ReturnCodeInternalError,
+			},
+		)
+		return
+	}
+
+	c.JSON(
```

Comment:
Perhaps we can use `shared.RespondWithSuccess` here (optional)?

Reply:
done